### PR TITLE
Add support for CommandCollection and click.Argument

### PR DESCRIPTION
**Pull request by beezz**

### CODE REVIEW

**Comment by untitaker** [comment]
Thanks, please add tests for the new command types though :)

**Comment by beezz** [comment]
Hi @untitaker , I added basic tests.

File: tests/test_command_collection.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    c = ClickCompleter(click.CommandCollection(
+        sources=[foo_group, foobar_group]
+    ))
+    completions = list(c.get_completions(Document(u'foo')))
+
+    assert set(x.text for x in completions) == set([
+        u'foo_cmd',
+        u'foobar_cmd'
+    ])
```
Comment (untitaker) [inline tests/test_command_collection.py]: If I revert your implementation using `git checkout master click_repl`, this test still passes.

**Comment by untitaker** [comment]
Sorry, I was confusing the test with something else. I merged everything but the last commit.

**Comment by beezz** [comment]
@untitaker actually you are right. The tests for command collection are wrong.

**Pull request closed without merging**